packages/app/pages/organization/course/add.tsx

```diff
@@ -0,0 +1,275 @@
+import {
+  Breadcrumb,
+  Button,
+  Card,
+  Col,
+  Form,
+  Input,
+  Row,
+  Select,
+  Spin,
+  message,
+} from "antd";
+import Head from "next/head";
+import { ReactElement } from "react";
+import NavBar from "../../../components/Nav/NavBar";
+import { StandardPageContainer } from "../../../components/common/PageContainer";
+import { useSemester } from "../../../hooks/useSemester";
+import { useRouter } from "next/router";
+import { useOrganization } from "../../../hooks/useOrganization";
+import { useProfile } from "../../../hooks/useProfile";
+import { OrganizationRole } from "@koh/common";
+import DefaultErrorPage from "next/error";
+import { API } from "@koh/api-client";
+import useSWR from "swr";
+
+export default function Add(): ReactElement {
+  const profile = useProfile();
+  const router = useRouter();
+  const { organization } = useOrganization(profile?.organization.id);
+  const isAdmin =
+    profile?.organization.organizationRole === OrganizationRole.ADMIN;
+  const isProfessor =
+    profile?.organization.organizationRole === OrganizationRole.PROFESSOR;
+
+  if (profile && !isAdmin && !isProfessor) {
```

Comment:
I think we should use useEffect to prevent race condition between loading page and fetching API?

Reply:
Can you give me an example, I don't know what you mean.

Reply:
```
useEffect(() => {
    if (code && !FAILED_CODES[Number(code)]) {
      router.push('/')
    }
  }, [code, router])
```